RecoBTag/DeepFlavour/interface/TrackInfoBuilder.h

```diff
@@ -0,0 +1,119 @@
+
+
+#include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "DataFormats/Candidate/interface/VertexCompositePtrCandidate.h"
+
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/IPTools/interface/IPTools.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "RecoVertex/VertexTools/interface/VertexDistance3D.h"
+#include "TVector3.h"
+
+namespace deep{
+
+// adapted from DeepNtuples
+class TrackInfoBuilder{
+public:
+    TrackInfoBuilder(edm::ESHandle<TransientTrackBuilder> & build):
+        builder(build),
+        trackMomentum_(0),
+        trackEta_(0),
+        trackEtaRel_(0),
+        trackPtRel_(0),
+        trackPPar_(0),
+        trackDeltaR_(0),
+        trackPtRatio_(0),
+        trackPParRatio_(0),
+        trackSip2dVal_(0),
+        trackSip2dSig_(0),
+        trackSip3dVal_(0),
+        trackSip3dSig_(0),
+
+        trackJetDistVal_(0),
+        trackJetDistSig_(0)
+{
+
+
+}
+
+    void buildTrackInfo(const reco::Candidate * candidate ,const math::XYZVector&  jetDir, GlobalVector refjetdirection, const reco::Vertex & pv){
```

Comment:
please add some docs.
I'm trying to understand why jetDir and refjetdirection are different:  why different arguments, and why different type?
having one type would at least reduce the amount of necessary gymnastics in the user code

Reply:
This is not my code, but from the training frameowork (I think either @mverzett or @jkiesele). I will look into it and ask them in case I do not get it.